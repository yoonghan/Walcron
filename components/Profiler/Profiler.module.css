.profiler {
  display: flex;
  justify-content: space-evenly;
}

@media only screen and (max-width: 640px) {
  .profiler {
    flex-direction: column;
  }
  .user:nth-child(even) {
    text-align: right
  }
}

.user {
  padding: 30px;
}

.name {
  font-weight: bold;
}

.divider {
  text-align: left;
}

@media only screen and (min-width: 641px) {
  .user {
    padding: 15px;
  }
  .name {
    font-weight: bold;
    text-align: center;
  }
  .divider {
    text-align: center;
  }
}
