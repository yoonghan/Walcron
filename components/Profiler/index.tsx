`use strict`

import * as React from "react";

export interface IProfiler {
  name: string;
  description: JSX.Element;
  imgSrc: string;
}

export interface ProfilerProps {
  profiles: Array<IProfiler>;
}

const _getUser = (idx: number, length: number, name: string, description: JSX.Element, imgSrc: string) => (
  <div className={"user"} key={"user_"+idx}>
    <img src={imgSrc} alt={name}/>
    <p className={"name"}>{name}</p>
    <div className={"divider"}>------</div>
    {description}
    <style jsx>
      {`
        .user {
          padding: 30px;
        }

        .name {
          font-weight: bold;
        }

        img {
          height: 200px;
          margin: auto;
          display: block;
        }

        @media only screen and (min-width: 641px) {
          .user {
            width: ${(100/length)}%;
            padding: 15px;
          }
          .name {
            font-weight: bold;
            text-align: center;
          }
          .divider {
            text-align: center;
          }
        }
      `}
    </style>
  </div>
)

const Profiler: React.SFC<ProfilerProps> = ({profiles}) => {
  return (
    <div className="profiler">
      {
        profiles.map((profile, idx) => (
          _getUser(idx, profiles.length, profile.name, profile.description, profile.imgSrc)
        ))
      }
      <style jsx>
        {`
          .profiler {
            display: flex;
            justify-content: space-evenly;
          }
          @media only screen and (max-width: 640px) {
            .profiler {
              flex-direction: column;
            }
          }
        `}
      </style>
    </div>
  );
}

export default Profiler;
