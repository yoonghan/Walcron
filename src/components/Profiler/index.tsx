import Image, { StaticImageData } from "next/image"
import styles from "./Profiler.module.css"

export interface IProfiler {
  name: string
  description: JSX.Element
  imgSrc: string | StaticImageData
  width?: number
  height?: number
}

export interface Props {
  profiles: Array<IProfiler>
}

const _getUser = (
  idx: number,
  name: string,
  description: JSX.Element,
  imgSrc: string | StaticImageData,
  width = 285,
  height = 385
) => (
  <div className={styles["user"]} key={"user_" + idx}>
    <Image src={imgSrc} alt={name} width={width} height={height} />
    <p className={styles.name}>{name}</p>
    <div className={styles["divider"]}></div>
    {description}
  </div>
)

const Profiler = ({ profiles }: Props) => {
  return (
    <div className={styles["profiler"]}>
      {profiles.map((profile, idx) =>
        _getUser(
          idx,
          profile.name,
          profile.description,
          profile.imgSrc,
          profile.width,
          profile.height
        )
      )}
    </div>
  )
}

export default Profiler
