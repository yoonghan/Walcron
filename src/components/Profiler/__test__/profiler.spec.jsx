import * as React from 'react';
import { shallow } from 'enzyme';
import Profiler from "../index";

describe('Profiler', () => {
  it('renders normally', () => {
    const wrapper = shallow(
      <Profiler
        profiles={[
          {
            name: "name1",
            description: <div/>,
            imgSrc: "imgSrc1"
          },
          {
            name: "name2",
            description: <div/>,
            imgSrc: "imgSrc2"
          }
        ]}
      />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  })
});
