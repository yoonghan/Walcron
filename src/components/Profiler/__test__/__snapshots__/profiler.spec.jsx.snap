// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profiler renders normally 1`] = `
"<div className=\\"jsx-2154105552 profiler\\">
  <div className=\\"jsx-2143176054 user\\">
    <img src=\\"imgSrc1\\" alt=\\"name1\\" className=\\"jsx-2143176054\\" />
    <p className=\\"jsx-2143176054 name\\">
      name1
    </p>
    <div className=\\"jsx-2143176054 divider\\">
      ------
    </div>
    <div />
    <JSXStyle id=\\"965410664\\" dynamic={{...}}>
      .user.__jsx-style-dynamic-selector{padding:30px;}.name.__jsx-style-dynamic-selector{font-weight:bold;}img.__jsx-style-dynamic-selector{height:200px;margin:auto;display:block;}@media only screen and (min-width:641px){.user.__jsx-style-dynamic-selector{width:50%;padding:15px;}.name.__jsx-style-dynamic-selector{font-weight:bold;text-align:center;}.divider.__jsx-style-dynamic-selector{text-align:center;}}
    </JSXStyle>
  </div>
  <div className=\\"jsx-2143176054 user\\">
    <img src=\\"imgSrc2\\" alt=\\"name2\\" className=\\"jsx-2143176054\\" />
    <p className=\\"jsx-2143176054 name\\">
      name2
    </p>
    <div className=\\"jsx-2143176054 divider\\">
      ------
    </div>
    <div />
    <JSXStyle id=\\"965410664\\" dynamic={{...}}>
      .user.__jsx-style-dynamic-selector{padding:30px;}.name.__jsx-style-dynamic-selector{font-weight:bold;}img.__jsx-style-dynamic-selector{height:200px;margin:auto;display:block;}@media only screen and (min-width:641px){.user.__jsx-style-dynamic-selector{width:50%;padding:15px;}.name.__jsx-style-dynamic-selector{font-weight:bold;text-align:center;}.divider.__jsx-style-dynamic-selector{text-align:center;}}
    </JSXStyle>
  </div>
  <JSXStyle id=\\"2154105552\\">
    .profiler.jsx-2154105552{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:space-evenly;-webkit-justify-content:space-evenly;-ms-flex-pack:space-evenly;justify-content:space-evenly;}@media only screen and (max-width:640px){.profiler.jsx-2154105552{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}}
  </JSXStyle>
</div>"
`;
